src/language/keywords.md

```diff
@@ -16,23 +16,23 @@ The following table lists the words that the Dart language treats specially.
 {% assign bii = '&nbsp;<sup title="built-in-identifier" alt="built-in-identifier">2</sup>' %}
 {% assign lrw = '&nbsp;<sup title="limited reserved word" alt="limited reserved word">3</sup>' %}
 <div class="table-wrapper" markdown="1">
-| [abstract][]{{bii}}   | [else][]                 | [import][]{{bii}}     | [show][]{{ckw}}    |
-| [as][]{{bii}}         | [enum][]                 | [in][]                | [static][]{{bii}}  |
-| [assert][]            | [export][]{{bii}}        | [interface][]{{bii}}  | [super][]          |
-| [async][]{{ckw}}      | [extends][]              | [is][]                | [switch][]         |
-| [await][]{{lrw}}      | [extension][]{{bii}}     | [late][]{{bii}}       | [sync][]{{ckw}}    |
-| [base][]{{bii}}       | [external][]{{bii}}      | [library][]{{bii}}    | [this][]           |
-| [break][]             | [factory][]{{bii}}       | [mixin][]{{bii}}      | [throw][]          |
-| [case][]              | [false][]                | [new][]               | [true][]           |
-| [catch][]             | [final (variable)][]     | [null][]              | [try][]            |
-| [class][]             | [final (class)][]{{bii}} | [on][]{{ckw}}         | [typedef][]{{bii}} |
-| [const][]             | [finally][]              | [operator][]{{bii}}   | [var][]            |
-| [continue][]          | [for][]                  | [part][]{{bii}}       | [void][]           |
-| [covariant][]{{bii}}  | [Function][]{{bii}}      | [required][]{{bii}}   | [when][]           |
-| [default][]           | [get][]{{bii}}           | [rethrow][]           | [while][]          |
-| [deferred][]{{bii}}   | [hide][]{{ckw}}          | [return][]            | [with][]           |
-| [do][]                | [if][]                   | [sealed][]{{bii}}     | [yield][]{{lrw}}   |
-| [dynamic][]{{bii}}    | [implements][]{{bii}}    | [set][]{{bii}}        |                    |
+| [abstract][]{{bii}}   | [else][]                  | [implements][]{{bii}} | [set][]{{bii}}     |
+| [as][]{{bii}}         | [enum][]                  | [import][]{{bii}}     | [show][]{{ckw}}    |
+| [assert][]            | [export][]{{bii}}         | [in][]                | [static][]{{bii}}  |
+| [async][]{{ckw}}      | [extends][]               | [interface][]{{bii}}  | [super][]          |
+| [await][]{{lrw}}      | [extension][]{{bii}}      | [is][]                | [switch][]         |
+| [base][]{{bii}}       | [extension type][]{{bii}} | [late][]{{bii}}       | [sync][]{{ckw}}    |
```

Comment:
(This marks both `extension` and `extension type` as built-in identifiers. Since `extension type` is not an identifier - it's two - it can't be a built-in identifier. I think `type` might be a contextual keyword, because it's only a keyword after `extension` when `extension` occurs as a declaration.)

Reply:
Hmm, not sure how to handle this. I could change it to just:

> `| [type][]{{ckw}} |`

Or

> `| [(extension) type][]{{ckw}} |`

The same way we have parens acround (class) and (variable) for `final`, but I'm not sure if that fixes it. @parlough

Reply:
`extension` is a built-in identifier, but `type` is not. You could say that `type` is a contextual built-in identifier (or a contextual reserved word, because the 'reserved word'/'built-in identifier' distinction makes no difference here, and we might want to describe `'type'` in the same way that we currently describe `'await'` and `'yield'`).

The required context is that we're parsing a `<topLevelDefinition>` and we have accepted the token `'extension'`. In that situation the token `'type'` is a reserved word that implies we're parsing an `<extensionTypeDeclaration>`, not an `<extensionDeclaration>`.

I think `'type'` could then have its own entry of the form `[type][]{{lrw}}`.

Reply:
Thank you! That's a tricky one but I'm glad we could get it right